onedrive/src/main/java/ch/cyberduck/core/onedrive/GraphSession.java

```diff
@@ -135,7 +137,9 @@ protected OneDriveAPI connect(final Proxy proxy, final HostKeyCallback key, fina
             @Override
             public void process(final HttpRequest request, final HttpContext context) throws HttpException, IOException {
                 if(request.containsHeader(HttpHeaders.AUTHORIZATION)) {
-                    super.process(request, context);
+                    if(context.getAttribute(REDIRECT_LOCATIONS) == null) {
+                        super.process(request, context);
+                    }
```

Comment:
Shouldn't there be an `else` with `request.removeHeaders(HttpHeaders.AUTHORIZATION);`? Otherwise the header is still there, potentially with expired tokens.